test/redfishpower.c

```diff
@@ -234,7 +234,8 @@ _prompt_loop(void)
                    || !strcmp(buf, "setstatpath")
                    || !strcmp(buf, "setonpath")
                    || !strcmp(buf, "setoffpath")
-                   || !strcmp(buf, "setcyclepath")) {
+                   || !strcmp(buf, "setcyclepath")
+                   || !strcmp(buf, "settimeout")) {
```

Comment:
Based on your note in #67 where on and off took very different amounts of time, does it make sense to have a per-command timeout rather than just one?

I can't tell from the code whether "cycle" is covered by this timeout?

Reply:
In redfishpower land, yes "cycle" is covered by this timeout.  But in powerman land, it depends if "cycle" is native or not.   
 If "cycle" is an "off" and then "on", then the timeout is for a single off/on.  powerman's 60 second timeout would encompass the entire "off" and "on" together.

hmmm ... interesting idea.  I'm going back and forth on it.   It's a good idea, but presumably we only need to cover the worst case timeout.  Since powerman would normally return faster for the faster cases anyways.

Also, i like the idea of making powerman's timeout and redfishpower's timeout the same, to make things easier for configuration.  But I see your point ... if you're modifying device files, maybe the user is advanced enough already?

Reply:
I tend to think all's fair in the device files as long as we document what's going on.

I think we have to make sure that a redfishpower timeout is interpreted in some way as failed completion by the script rather than failure to produce expected output as the latter would cause the powerman timeout to kick in and redfishpower to be restarted, only after it expires.  I know we can fail a status query right away but I'm not so sure about an on/off operation.

Reply:
Ideally there would be no timeouts but if we are going to occasionally fail I think we want to do it as fast as possible so powerman can become responsive again...

Reply:
> I think we have to make sure that a redfishpower timeout is interpreted in some way as failed completion by the script rather than failure to produce expected output 

Right now it'll output something like `node1: timeout`, which is what I did in ipmipower.  Would that trigger the scriptlet in the expected way?

Reply:
I am not sure we have any way to fail a power operation other than to let the powerman timeout expire.  E.g. if you return "timeout" and it is not expected, that will eventually fail.  But it won't fail fast.  I'll open a bug.